AEPLifecycle/Sources/LifecycleV2/LifecycleV2.swift

```diff
@@ -65,10 +82,52 @@ class LifecycleV2 {
         stateManager.update(state: .PAUSE) { [weak self] (updated: Bool) in
             guard let self = self else { return }
             guard updated else { return }
+            guard let xdm = self.xdmMetricsBuilder.buildAppCloseXDMData(launchDate: self.dataStoreCache.getAppStartDate() ?? Date(), closeDate: pauseDate, isCloseUnknown: false) else { return }
```

Comment:
Couple of things that need to be updated here:
- the pauseDate needs to be persisted with dataStoreCache.setAppPauseDate
- buildAppCloseXDMData seems to require optional Date for launchDate and closeDate and handle those appropriately, we should not pass current Date() as app start date in this situation as this impacts the session length computation.

Reply:
Yes I wanted to discuss this, so we should return nil if nil launch date found?

Reply:
We can pass nil to buildAppCloseXDMData -> computeAppCloseData. SessionLength should be 0 if it cannot be computed because of a read fail from persistence.